packages/vision/src/helpers.js

```diff
@@ -73,14 +73,27 @@ var coerceImage = (image, callback) => {
  * @return {Function} - The function that, when called, will call annotateImage
  *   asking for the single feature annotation.
  */
-var _createSingleFeatureMethod = featureValue => {
-  return function(image, options) {
-    return this.annotateImage({
-      image: image,
-      features: [{type: featureValue}],
-    }, options);
-  };
-};
+ var _createSingleFeatureMethod = featureValue => {
+   return function(annotateImageRequest, callOptions) {
+     annotateImageRequest.features = annotateImageRequest.features || [{
+       type: featureValue,
+     }];
+
+     // If the user submitted explicit features that do not line up with
+     // the precise method called, throw an exception.
+     for (let feature of annotateImageRequest.features) {
+       if (feature.type !== featureValue) {
+         throw new Error(
```

Comment:
I do not believe Node v4 supports `throw` closing the iterator in a for..of loop. Check [node.green](http://node.green/). May just want to use a standard for loop, or `annotateImageRequest.features.forEach`.

Reply:
My original reaction was, "But the unit tests passed," but it occurs to me that `assert.throws` is pretty broad. Happy to change this. :-)

Reply:
It does work in Node 4:

```
lukesneeringer@lukesneeringer0:~/Code/Google/google-cloud-node:issue-2553$ nvm use 4
Now using node v4.8.4 (npm v2.15.11)
lukesneeringer@lukesneeringer0:~/Code/Google/google-cloud-node:issue-2553$ node
> for (let i of [1, 2, 3, 4, 5]) {
... console.log(i)
... if (i == 3) {
..... throw new Error('foo')
..... }
... }
1
2
3
Error: foo
    at repl:5:7
    at REPLServer.defaultEval (repl.js:272:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:441:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:203:10)
    at REPLServer.Interface._line (readline.js:542:8)
    at REPLServer.Interface._ttyWrite (readline.js:819:14)
```